File: Item.rb
class Item

	attr_reader :name, :type, :stength_required
	attr_reader :parameters

	@@items = {}

	def initialize(name, type, stength_required = 0, parameters)
		@name = name
		@type = type
		@stength_required = stength_required
		@damage = parameters[:damage]
		@defence = parameters[:defence]
		@hitpoints = parameters[:hitpoints]
		@magic_defence = parameters[:magic_defence]
		@parameters = parameters
		@@items[name.downcase.gsub(/\s+/, '_').to_sym] = {name: name, type: type, stength_required: stength_required, parameters: parameters}
	end

	def self.new_item(name, type, stength_required=0, parameters)
		Item.new(name, type, stength_required=0, parameters)
	end

	def self.item_list
		@@items.each {|key, value| p "<<key>>: #{key}, <<value>>: #{value}."}
	end

	def self.get_item(name)
		@@items[name]
	end

	private

end

File: src/game_logic/modules/item.js
const items = {
	necklace: new Map(),
	helmet: new Map(),
	ring: new Map(),
	weapon: new Map(),
	armor: new Map(),
	shield: new Map(),
	gloves: new Map(),
	leggins: new Map(),
	boots: new Map(),
	none: new Map()
};
// example
// items = {
// 	weapon: {"nunjucks": itemInstance.params{
// damage: 5, hitpoints: 3
// }}
// }

var itemList = function(){
	console.log(items);
}

export default class Item {
	constructor(params){
		this.params = params;
	}

	static getAll(){
		for ( let [name, map] of Object.entries(items) ) {
			map.forEach(function(itemInst, key){
				console.log(`${key}: ${itemInst.params.type}`);
			})
		}		
	}

	static newItem(params){
		if (!items[params.type]){
			console.log("There is no such type. " + params.name);
		}

		items[params.type].set(params.name, new Item(params));
	}

}

//weapon
Item.newItem({name: "Axe of Executioner", type: "weapon", hitpoints: 80, damage: 75, critChance: 10, url:"./images/items/weapon/AxeofExecutioner.jpg", description: "Legendary weapon of the executioner who has killed his own son during execution. It makes his owner incredibly strong."});
Item.newItem({name: "Makarov Pistol", type: "weapon", damage: 50, critChance: 15, critDamage: 100, url:"./images/items/weapon/MakarovPistol.png", description: "Reliable and easy in maintain pistol."});
Item.newItem({name: "Iron Axe", type: "weapon", damage: 7, url:"./images/items/weapon/IronAxe.png", description: "Simple weapon. Low damage."});
//armor
Item.newItem({name: "Decrepit helmet", type: "helmet", hitpoints: 10, defence: 5});
Item.newItem({name: "Dulle Dagger", type: "weapon", damage: 5});
Item.newItem({name: "Metal Cuirass", type: "armor", hitpoints: 50, defence: 25});
Item.newItem({name: "Shining Shield", type: "shield", hitpoints: 150, defence: 50, magicDefence: 30, damage: 10});

